docs/reference/setup/advanced-configuration.asciidoc

```diff
@@ -1,55 +1,63 @@
 
-If you need to override the default <<heap-size-settings,heap size settings>>,
-follow the best practices below.
+NOTE: Automatic heap sizing requires the <<jvm-version,bundled JDK>> or, if using
+a custom JRE location, a Java 14 or later JRE.
 
-{es} assigns the entire heap specified in
-<<jvm-options,jvm.options>> via the `Xms` (minimum heap size) and `Xmx` (maximum
-heap size) settings. These two settings must be equal to each other.
+To override the default heap size, set the minimum and maximum heap size
+settings, `Xms` and `Xmx`. The minimum and maximum values must be the same.
 
-The value for these settings depends on the amount of RAM available on your
-server:
+The heap size should be based on the available RAM:
 
-* Set `Xmx` and `Xms` to no more than 50% of your total system memory. {es} requires
-memory for purposes other than the JVM heap and it is important to leave
-space for this. For instance, {es} uses off-heap buffers for efficient
-network communication, relies on the operating system's filesystem cache for
-efficient access to files, and the JVM itself requires some memory too. It is
-normal to observe the {es} process using more memory than the limit
+* Set `Xms` and `Xmx` to no more than 50% of your total memory. {es} requires
+memory for purposes other than the JVM heap. For example, {es} uses
+off-heap buffers for efficient network communication and relies
+on the operating system's filesystem cache for
+efficient access to files. The JVM itself also requires some memory. It is
+normal for the {es} process to use more memory than the limit
 configured with the `Xmx` setting.
++
+NOTE: When running in a container, such as <<docker,Docker>>, total memory is
+defined as the amount of memory visible to the container, not the total system
+memory on the host.
 
-* Set `Xmx` and `Xms` to no more than the threshold that the JVM uses for
-compressed object pointers (compressed oops). The exact threshold varies but
-is near 32 GB. You can verify that you are under the threshold by looking for a line in the logs like the following:
+* Set `Xms` and `Xmx` to no more than the threshold that the JVM uses for
+compressed ordinary object pointers (oops). The exact threshold varies but
+is near 32 GB. You can verify that you are under the threshold by checking
+the logs for a heap size entry like this:
 +
 [source,txt]
 ----
 heap size [1.9gb], compressed ordinary object pointers [true]
 ----
 
-* Set `Xmx` and `Xms` to no more than the threshold for zero-based
+* Set `Xms` and `Xmx` to no more than the threshold for zero-based
 compressed oops. The exact threshold varies but 26GB is safe on most
 systems and can be as large as 30GB on some systems. You can verify that
 you are under this threshold by starting {es} with the JVM options
-`-XX:+UnlockDiagnosticVMOptions -XX:+PrintCompressedOopsMode` and looking for
-a line like the following:
+`-XX:+UnlockDiagnosticVMOptions -XX:+PrintCompressedOopsMode` and checking
```

Comment:
I saw https://github.com/elastic/elasticsearch/issues/71644 and tested this locally. It looks like `-XX:+UnlockDiagnosticVMOptions -XX:+PrintCompressedOopsMode` is no longer supported.

We may want to update this section based on that ticket. We can also handle as a separate issue if wanted.

Reply:
I'll handle as a follow up.